### ISSUE TITLE
Could not initialize NNPACK

### ISSUE THREAD

**Issue by chen003** [lukas-blecher/LaTeX-OCR#74]
Hi Lukas, I had the following problem when running pix2tex.py:
```
Predict LaTeX code for image ("?"/"h" for help). 
[W NNPACK.cpp:79] Could not initialize NNPACK! Reason: Unsupported hardware.
\frac{\mathbb{R}\mathbb{R}}{\omega_{i}^{\ast}}\lambda 
```
The returned formula is completely wrong. 

Another problem is I can't use gui.py, I got the following error message. This is probably unrelated to the first one.
```
Traceback (most recent call last):
  File "gui.py", line 11, in <module>
    from pynput.mouse import Controller
  File "/Users/xingyanchen/opt/anaconda3/lib/python3.8/site-packages/pynput/__init__.py", line 40, in <module>
    from . import keyboard
  File "/Users/xingyanchen/opt/anaconda3/lib/python3.8/site-packages/pynput/keyboard/__init__.py", line 31, in <module>
    backend = backend(__name__)
  File "/Users/xingyanchen/opt/anaconda3/lib/python3.8/site-packages/pynput/_util/__init__.py", line 70, in backend
    return importlib.import_module('._' + module, package)
  File "/Users/xingyanchen/opt/anaconda3/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/xingyanchen/opt/anaconda3/lib/python3.8/site-packages/pynput/keyboard/_darwin.py", line 29, in <module>
    import Quartz
  File "/Users/xingyanchen/opt/anaconda3/lib/python3.8/site-packages/Quartz/__init__.py", line 110, in <module>
    _load()
  File "/Users/xingyanchen/opt/anaconda3/lib/python3.8/site-packages/Quartz/__init__.py", line 28, in _load
    from Quartz import CoreGraphics as m
  File "/Users/xingyanchen/opt/anaconda3/lib/python3.8/site-packages/Quartz/CoreGraphics/__init__.py", line 147, in <module>
    _load(mod)
  File "/Users/xingyanchen/opt/anaconda3/lib/python3.8/site-packages/Quartz/CoreGraphics/__init__.py", line 55, in _load
    import Quartz.CoreGraphics._callbacks as m
RuntimeError: Wrong version of PyObjC C API (got 22, expected 21)
```
My OS is macOS Monterey. I have no problem installing the packages from requirements.txt. I installed Node.js and ImageMagick from homebrew.

Thanks for reading this!

**Comment by chen003** [comment]
Could it be due to the apple m1 chip? I'm using a macbook pro with m1x.

**Comment by lukas-blecher** [comment]
I'm not sure, maybe. The error message does mention hardware.
Can you share the input image so I can check what the expected output would be?

**Comment by chen003** [comment]
<img width="331" alt="Screen Shot 2022-01-05 at 5 29 37 PM" src="https://user-images.githubusercontent.com/11232569/148304284-b74fa3b8-8108-4731-9c9a-11941f9b0a2f.png">
It's just a screen shot from the example page https://lukas-blecher.github.io/LaTeX-OCR/performance

**Comment by lukas-blecher** [comment]
Thanks for brining it to my attention. I think the latest resizer network did a poor job on the input image
![image](https://user-images.githubusercontent.com/55287601/148643631-fc9345df-eaef-4c30-8680-06508cec338c.png)

Can you confirm the image above returns `\nabla\times\vec{\mathrm{B}}-\frac{1}{C}\frac{\delta\vec{\mathrm{E}}}{\delta t}=4\pi\rho`? (right click and copy the image above, then run `pix2tex`)

For your GUI problem you might want to install another version of `pyobjc` (https://pyobjc.readthedocs.io/en/latest/install.html)

**Comment by chen003** [comment]
Indeed the image works.

`python /Users/xingyanchen/Documents/Git/softare/LaTeX-OCR/pix2tex.py 
/Users/xingyanchen/opt/anaconda3/lib/python3.8/site-packages/torch/nn/functional.py:718: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  ../c10/core/TensorImpl.h:1156.)
  return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)
Predict LaTeX code for image ("?"/"h" for help). 
[W NNPACK.cpp:79] Could not initialize NNPACK! Reason: Unsupported hardware.
\nabla\times\vec{\mathrm{B}}-\frac{1}{C}\frac{\delta\vec{\mathrm{E}}}{\delta t}=4\pi\rho `

**Comment by chen003** [comment]
I'm wondering why the package work without loading the NNPACK? 
So the three problems I have are independent: 1) can't load NNPACK 2) image resizer fails as you mentioned 3) GUI problem due to pyobj. That's a bit funny :)

**Comment by lukas-blecher** [comment]
I think the issue might have been solved by #105
I think the warning about NNPACK never was the reason for the bad output. 
Sometimes the model just acts up
